### PR TITLE
REGR: get_group raising with axis=1

### LINKED ISSUE #54858: grp.get_group: IndexError: positional indexers are out-of-bounds
Reproducer:
```python
import pandas
pandas_df = pandas.DataFrame(
     {
         "col1": [0, 3, 2, 3],
         "col2": [4, 1, 6, 7],
         "col3": [3, 8, 2, 10],
         "col4": [1, 13, 6, 15],
         "col5": [-4, 5, 6, -7],
     }
)
pandas_groupby = pandas_df.groupby(axis=1, by=[1,2,3,2,1]) 
print(pandas_groupby.groups)  # <- {1: ['col1', 'col5'], 2: ['col2', 'col4'], 3: ['col3']}
pandas_groupby.get_group(1)  # <- IndexError: positional indexers are out-of-bounds
```

pandas 2.1.0rc0

```bash
>>> pandas_groupby.get_group(1)    
Traceback (most recent call last):
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\indexing.py", line 1675, in _get_list_axis
    return self.obj._take_with_is_copy(key, axis=axis)
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\generic.py", line 4088, in _take_with_is_copy
    result = self.take(indices=indices, axis=axis)
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\generic.py", line 4068, in take
    new_data = self._mgr.take(
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\internals\managers.py", line 874, in take
    indexer = maybe_convert_indices(indexer, n, verify=verify)
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\indexers\utils.py", line 282, in maybe_convert_indices
    raise IndexError("indices are out-of-bounds")
IndexError: indices are out-of-bounds

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\groupby\groupby.py", line 1060, in get_group
    return self._selected_obj.iloc[inds]
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\indexing.py", line 1152, in __getitem__
    return self._getitem_axis(maybe_callable, axis=axis)
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\indexing.py", line 1704, in _getitem_axis
    return self._get_list_axis(key, axis=axis)
  File "Miniconda3\envs\modin\lib\site-packages\pandas\core\indexing.py", line 1678, in _get_list_axis
    raise IndexError("positional indexers are out-of-bounds") from err
IndexError: positional indexers are out-of-bounds
```

### PR DESCRIPTION
**Pull request by phofl**
- [ ] closes #54858 (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

File: pandas/core/groupby/groupby.py [hunk 1]
```diff
@@ -1079,7 +1079,8 @@ def get_group(self, name, obj=None) -> DataFrame | Series:
             raise KeyError(name)
 
         if obj is None:
-            return self._selected_obj.iloc[inds]
+            indexer = inds if self.axis == 0 else (slice(None), inds)
```
Comment (mroeschke) [inline pandas/core/groupby/groupby.py:1082]: Just to confirm that we won't need this once `self.axis` is deprecated?

**Review by rhshadrach — APPROVED** [review]
lgtm

File: pandas/core/groupby/groupby.py (hunk 1 above)
Comment (phofl) [reply pandas/core/groupby/groupby.py:1082]: Removed, but yes

**Review by mroeschke — APPROVED** [review]

**Comment by phofl** [comment]
@meeseeksdev backport 2.1.x

**Pull request merged**